### PR TITLE
Add a simple server and client

### LINKED ISSUE #3: Introduce travis CI and check style tool

### LINKED ISSUE #4: Add simple kV storage
@kairbon

### PR DESCRIPTION
**Pull request by kairbon**
this close #4 
close #3

### CODE REVIEW

File: client/src/main/java/org/dst/dmeta/client/DmetaClient.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import java.util.concurrent.TimeoutException;
+
+public class DmetaClient {
+
+  private BoltCliClientService cliClientService = null;
+
+  public static final String groupId = "KV";
+
+  public static final String confStr = "127.0.0.1:8081,127.0.0.1:8082,127.0.0.1:8083";
```
Comment (jovany-wang) [inline client/src/main/java/org/dst/dmeta/client/DmetaClient.java]: hardcode?

File: server/src/main/java/org/dst/dmeta/server/DmetaStateMachine.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      iter.next();
+    }
+  }
+
+  @Override
+  public void onSnapshotSave(final SnapshotWriter writer, final Closure done) {
+    final ConcurrentHashMap<String, String> currVal = this.getStore();
+    Utils.runInThread(() -> {
+      final DmetaSnapshotFile snapshot =
+          new DmetaSnapshotFile(writer.getPath() + File.separator + "data");
```
Comment (jovany-wang) [inline server/src/main/java/org/dst/dmeta/server/DmetaStateMachine.java]: Let's add a TODO comment here to indicates that we should use rocksdb to do snapshot later.

File: client/src/main/java/org/dst/dmeta/client/DmetaClient.java (hunk 1 above)
Comment (kairbon) [reply client/src/main/java/org/dst/dmeta/client/DmetaClient.java]: This is just a TestUtil class. so i didn't do anything to configure

File: server/src/main/java/org/dst/dmeta/server/DmetaStateMachine.java (hunk 2 above)
Comment (kairbon) [reply server/src/main/java/org/dst/dmeta/server/DmetaStateMachine.java]: ok, i will do it

Comment (kairbon) [reply server/src/main/java/org/dst/dmeta/server/DmetaStateMachine.java]: I add it in DmetaSnapshotFile class. Because It is a implementation class

**Pull request merged**